File: Selenium/frameworkPytest/tests/productDetailPage.py
from selenium.webdriver.common.by import By
from utils.utils import Utils

class ProductDetailPage(Utils):
  def __init__(self, driver):
    super().__init__(driver)
    self.driver = driver

  def validateProductDetailInfo(self, productName, productPrice):
    # Validate product information
    assert self.driver.find_element(By.CSS_SELECTOR, ".product-information span span").text == productPrice
    productDetailName = self.driver.find_element(By.CSS_SELECTOR, ".product-information h2").text.strip()
    assert productDetailName.replace(" ", "") == productName.replace(" ", "")
    productInfo = ["Category:", "Availability", "Condition", "Brand"]

    productDetail = self.driver.find_elements(By.CSS_SELECTOR, ".product-information p")
    index = 0
    for item in productDetail:
      assert item.text, productInfo[index]
      index += 1

    # assert self.driver.find_elements(By.CSS_SELECTOR, ".product-information p")[0], "Category:"
    # assert self.driver.find_elements(By.CSS_SELECTOR, ".product-information p")[1], "Availability"
    # assert self.driver.find_elements(By.CSS_SELECTOR, ".product-information p")[2], "Condition"
    # assert self.driver.find_elements(By.CSS_SELECTOR, ".product-information p")[3], "Brand"

  def validateBrandName(self, brandName):
    brandInfo = self.driver.find_elements(By.CSS_SELECTOR, ".product-information p")[-1].text
    assert brandName in brandInfo.upper()

  def getProductNamePrice(self):
    productDetailName = self.driver.find_element(By.CSS_SELECTOR, ".product-information h2").text
    productDetailPrice = self.driver.find_element(By.CSS_SELECTOR, ".product-information span span").text
    return productDetailName, productDetailPrice
  
  def getCategory(self):
    productDetailCategory = self.driver.find_elements(By.CSS_SELECTOR, ".product-information p")[0].text
    return productDetailCategory
  
  def addMultipleItems(self, quantity):
    productDetailQuantity = self.driver.find_element(By.CSS_SELECTOR, ".product-information input#quantity")
    productDetailQuantity.clear()
    productDetailQuantity.send_keys(str(quantity))

  def addToCart(self):
    self.driver.find_element(By.CSS_SELECTOR, "button.cart").click()

File: playwright/pageObjects/productDetailPage.js
import { expect } from '@playwright/test';

export class ProductDetailPage {
  constructor(page) {
    this.page = page;
  }
  async validateProductDetailInfo(productName, productPrice) {
    await expect(this.page.locator(".product-information span span")).toHaveText(productPrice);
    await expect(this.page.locator(".product-information h2")).toHaveText(productName);

    const productInfo = ["Category:", "Availability", "Condition", "Brand"];
    let index = 0;
    for (const productDetail of await this.page.locator(".product-information p").all()){
      await expect(productDetail).toContainText(productInfo[index]);
      index += 1;
    };
    // To validate without for() ->
    // await expect(this.page.locator(".product-information p").first()).toContainText("Category:");
    // await expect(this.page.locator(".product-information p").nth(1)).toContainText("Availability");
    // await expect(this.page.locator(".product-information p").nth(2)).toContainText("Condition");
    // await expect(this.page.locator(".product-information p").last()).toContainText("Brand");
  }

  async validateBrandName(brandName) {
    expect(await this.page.locator(".product-information p").last().textContent()).toContain(brandName);
  }

  async getProductNamePrice(){
    const productDetailName = await this.page.locator(".product-information h2").textContent();
    const productName = productDetailName.replaceAll("  ", " ");
    const productDetailPrice = await this.page.locator(".product-information span span").textContent();
    return {productName, productDetailPrice};
  }

  async getCategory(){
    const productDetailCategory = await this.page.locator(".product-information p").first().textContent();
    const productCategory = productDetailCategory.replaceAll("  ", " ");
    return productCategory;
  }

  async addMultipleItems(quantity){
    const productDetailQuantity = this.page.locator(".product-information input#quantity")
    await productDetailQuantity.clear();
    await productDetailQuantity.fill(quantity.toString())
  }

  async addToCart(){
    await this.page.locator("button.cart").click();
  }
}